From 31197ad85957dd44656d555e9bd72de81d8c8341 Mon Sep 17 00:00:00 2001
From: Zev Weiss <zev@bewilderbeest.net>
Date: Thu, 24 Sep 2020 21:44:48 +0000
Subject: [PATCH 1/2] peci: add support for Coffee Lake Refresh Xeon E hardware

I'm not certain of some of the numbers for various hardware parameters,
but for our use I don't think it'll matter if they're wrong (our Xeons
don't provide DIMM temperatures or CPU core counts via PECI, so we'll
be skipping the former and hard-coding the latter).

---
 drivers/mfd/intel-peci-client.c       | 6 ++++++
 include/linux/mfd/intel-peci-client.h | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/mfd/intel-peci-client.c b/drivers/mfd/intel-peci-client.c
index 24f15438634c..52c694f18287 100644
--- a/drivers/mfd/intel-peci-client.c
+++ b/drivers/mfd/intel-peci-client.c
@@ -48,6 +48,12 @@ static const struct cpu_gen_info cpu_gen_info_table[] = {
 		.core_max      = CORE_MAX_ON_SKXD,
 		.chan_rank_max = CHAN_RANK_MAX_ON_SKXD,
 		.dimm_idx_max  = DIMM_IDX_MAX_ON_SKXD },
+	{ /* Coffee Lake Refresh Xeon E */
+		.family        = INTEL_FAM6,
+		.model         = INTEL_FAM6_COFFEELAKE_XE,
+		.core_max      = CORE_MAX_ON_CFLRE,
+		.chan_rank_max = CHAN_RANK_MAX_ON_CFLRE,
+		.dimm_idx_max  = DIMM_IDX_MAX_ON_CFLRE },
 };
 
 static int peci_client_get_cpu_gen_info(struct peci_client_manager *priv)
diff --git a/include/linux/mfd/intel-peci-client.h b/include/linux/mfd/intel-peci-client.h
index 7668d0cfa843..fedd692cd35a 100644
--- a/include/linux/mfd/intel-peci-client.h
+++ b/include/linux/mfd/intel-peci-client.h
@@ -18,6 +18,7 @@
 #define INTEL_FAM6_BROADWELL_X		0x4F
 #define INTEL_FAM6_SKYLAKE_X		0x55
 #define INTEL_FAM6_SKYLAKE_XD		0x56
+#define INTEL_FAM6_COFFEELAKE_XE	0x9e
 #endif
 
 #define INTEL_FAM6             6 /* P6 (Pentium Pro and later) */
@@ -38,6 +39,11 @@
 #define CHAN_RANK_MAX_ON_SKXD  2  /* Max number of channel ranks on Skylake D */
 #define DIMM_IDX_MAX_ON_SKXD   2  /* Max DIMM index per channel on Skylake D */
 
+/* FIXME: what should these numbers actually be? */
+#define CORE_MAX_ON_CFLRE      16 /* Max number of cores on Coffee Lake */
+#define CHAN_RANK_MAX_ON_CFLRE 2  /* Max number of channel ranks on Coffee Lake */
+#define DIMM_IDX_MAX_ON_CFLRE  2  /* Max DIMM index per channel on Coffee Lake */
+
 #define CORE_NUMS_MAX          CORE_MAX_ON_SKX
 #define CHAN_RANK_MAX          CHAN_RANK_MAX_ON_HSX
 #define DIMM_IDX_MAX           DIMM_IDX_MAX_ON_HSX
-- 
2.7.4

